### ISSUE TITLE
Hide "All Day" checkbox

### ISSUE THREAD

**Issue by stueja** [meeting-room-booking-system/mrbs-code#3725]
**Describe the problem you're facing/question you have**
I am trying to switch from period-based booking to time-based booking. Users shall, if necessary, select a start time and an end time.
In this scenario, the "all day" checkbox is not necessary. I read `areadefaults.inc.php` and `systemdefaults.inc.php` and could not wrap my head around a possibility to get rid of the checkbox. Is there a possibility which I have overlooked?

Also, setting the checkbox shows a strange behaviour. If the checkbox is activated, the times are garbled. Please see attached screenshots for settings and behaviour.

**Browser details (please complete the following information):**

**Server details (please complete the following information):**
 - MRBS version [e.g. 1.9.4]
 - Web server: [e.g. Apache 2.4.48]
 - OS: [e.g. CentOS 7.9]
 - PHP version: [e.g. 7.4.23]
 - Database used: [e.g. MySQL 8.0.26]

**Additional context**
Settings:
<img width="535" alt="Bildschirmfoto 2024-08-16 um 12 13 01" src="https://github.com/user-attachments/assets/af8156ab-fa3c-45fc-a221-c822151d7ba1">
Checkbox unset:
<img width="595" alt="Bildschirmfoto 2024-08-16 um 12 11 54" src="https://github.com/user-attachments/assets/fac55494-e72f-4088-988f-3ffdade35788">
Checkbox activated:
<img width="562" alt="Bildschirmfoto 2024-08-16 um 12 12 04" src="https://github.com/user-attachments/assets/916698fa-ef24-436a-be83-b1fabb30aaef">

**Comment by campbell-m** [comment]
> In this scenario, the "all day" checkbox is not necessary. I read areadefaults.inc.php and systemdefaults.inc.php and could not wrap my head around a possibility to get rid of the checkbox. Is there a possibility which I have overlooked?

No, I'm afraid there isn't a config setting, so you'd have to modify the MRBS code.  Adding a config setting would be easy though, but I'd be interested to understand why you don't want the checkbox.

> If the checkbox is activated, the times are garbled.

I couldn't see anything wrong in the screenshots.  Can you explain some more?

**Comment by stueja** [comment]
> No, I'm afraid there isn't a config setting, so you'd have to modify the MRBS code. Adding a config setting would be easy though, but I'd be interested to understand why you don't want the checkbox.

If I understood the settings right, then I set MRBS up to book 1 day now. In the screenshot from Aug 5, 9am to Aug 6, 9am. The user now only needs to 
  - call up the calendar element and pick another day to book the room for, say, 5 days.
  - and/or call up the time selector to pick a different amount of hours.
In this case, where 1 day is preset, the checkbox feels a little duplicate and inexplicable(tm) to the users.

> I couldn't see anything wrong in the screenshots. Can you explain some more?

With pleasure. The screenshot in which "All Day" ("Ganztägig" in the screenshot) is unchecked, the start time is 9:00am and the end time is 9:00am, showing "1 Day" (1 Tag) as duration which seems correct.

The second screenshot which has "All Day" checked, the start time changed to 8:00am, but the end time is 9:00pm (21:00) which is then not "all day" but "all day plus 13 hours".

I should have mentioned that I did not change the start time (to 8:00am) nor the end time (to 21:00) manually. MRBS did that once I checked the "all day" box.

**Comment by campbell-m** [comment]
Ah, I think there's a misunderstanding about what the All Day checkbook does.  It's a quick way of selecting all day, every day. In your example you had set the start date to 5 Aug and the end date to 6 Aug.  When you check the All Day checkbox then the booking is automatically adjusted to go from the start of the first slot on the start date to the end of the last slot on the end date.

**Comment by campbell-m** [comment]
Maybe the German translation is confusing? It's supposed to mean "all of one day", not "a 24 hour period, possibly spanning two days".
